File: backend/src/main/java/com/example/starter_project_2025/system/classes/entity/TrainingClass.java
package com.example.starter_project_2025.system.classes.entity;

import com.example.starter_project_2025.system.course_class.entity.CourseClass;
import com.example.starter_project_2025.system.learning.entity.Enrollment;
import com.example.starter_project_2025.system.semester.entity.Semester;
import com.example.starter_project_2025.system.user.entity.User;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "classes")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class TrainingClass {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "class_id")
    private UUID id;

    @Column(name = "class_name")
    private String className;

    @OneToMany(mappedBy = "classInfo")
    @JsonManagedReference
    private List<CourseClass> courseClasses;

    @OneToMany(mappedBy = "trainingClass")
    @JsonManagedReference
    private List<Enrollment> enrollments;

    @Column(name = "class_code", unique = true)
    private String classCode;
    @Column(name = "enrollment_key", length = 500)
    private String enrollmentKey;

    @Enumerated(EnumType.STRING)
    private ClassStatus classStatus;

    @Column(name = "is_active")
    private Boolean isActive = false;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id")
    private User creator;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "semester_id", nullable = false)
    @JsonBackReference
    private Semester semester;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "approver_id")
    private User approver;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;
}

File: frontend/src/types/trainingClass.ts
/* ========================
   Training Class Types
======================== */

import type { ClassStatusType } from "@/pages/training-classes/enum/ClassStatus";

export interface TrainingClass {
  id: string;
  className: string;
  description?: string;
  classCode: string;
  enrollmentKey?: string;
  isActive: boolean;
  creatorName?: string;
  approverName?: string;
  semesterName?: string;
  trainingProgramId?: string;
  trainingProgramName?: string;
  startDate: string;
  endDate: string;
  status: ClassStatusType;
  trainerNames?: string[];
}

export interface CreateTrainingClassRequest {
  className: string;
  classCode: string;
  semesterId: string;
  trainingProgramId: string;
  startDate: string;
  endDate: string;
}

export interface UpdateTrainingClassRequest {
  className?: string;
  classCode?: string;
  semesterId?: string;
  startDate?: string;
  endDate?: string;
  description?: string;
}

/* ========================
   Semester Types
======================== */

export interface SemesterResponse {
  id: string;
  name: string;
  startDate: string;
  endDate: string;
}

export interface TrainingClassSemesterResponse {
  semesterId: string;
  semesterName: string;
  classes: TrainingClass[];
}